x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java

```diff
@@ -304,26 +383,52 @@ private ExtractedFields detectExtractedFields(Set<String> fields, Set<FieldSelec
         return extractedFields;
     }
 
-    private ExtractedFields deduplicateMultiFields(ExtractedFields extractedFields, boolean preferSource,
+    private ExtractedFields deduplicateMultiFields(ExtractedFields extractedFields,
+                                                   boolean preferSource,
                                                    Set<FieldSelection> fieldSelection) {
-        Set<String> requiredFields = config.getAnalysis().getRequiredFields().stream().map(RequiredField::getName)
+        Set<String> requiredFields = config.getAnalysis()
+            .getRequiredFields()
+            .stream()
+            .map(RequiredField::getName)
             .collect(Collectors.toSet());
+        Set<String> processorInputFields = extractedFields.getProcessedFieldInputs();
         Map<String, ExtractedField> nameOrParentToField = new LinkedHashMap<>();
         for (ExtractedField currentField : extractedFields.getAllFields()) {
             String nameOrParent = currentField.isMultiField() ? currentField.getParentField() : currentField.getName();
             ExtractedField existingField = nameOrParentToField.putIfAbsent(nameOrParent, currentField);
             if (existingField != null) {
                 ExtractedField parent = currentField.isMultiField() ? existingField : currentField;
                 ExtractedField multiField = currentField.isMultiField() ? currentField : existingField;
+                // If required fields contains parent or multifield and the processor input fields reference the other, that is an error
+                // we should not allow processing of data that is required.
+                if ((requiredFields.contains(parent.getName()) && processorInputFields.contains(multiField.getName()))
+                    || (requiredFields.contains(multiField.getName()) && processorInputFields.contains(parent.getName()))) {
+                    throw ExceptionsHelper.badRequestException(
+                        "feature_processors cannot be applied to required fields for analysis multi-field [{}] parent [{}]",
+                        multiField.getName(),
+                        parent.getName());
+                }
+                // If processor input fields have BOTH, we need to keep both.
```

Comment:
Doesn't that lead to complexity? Could we error here and request that a field is always referred to in the same way?

Reply:
For sure, we can error here. It would be weird that the same user chooses separate multi-fields for different processed fields.